### PR TITLE
OS-8180 lx want support for NoNewPrivs

### PR DESCRIPTION
**Pull request by papertigers**

### CODE REVIEW

File: usr/src/uts/common/brand/lx/syscall/lx_prctl.c
Comment (jasonbking) [inline usr/src/uts/common/brand/lx/syscall/lx_prctl.c:281]: Since we're not declaring any new local variables, we don't need the new scope.

Comment (papertigers) [reply usr/src/uts/common/brand/lx/syscall/lx_prctl.c:281]: I was just matching existing style such as with `LX_PR_GET_SECUREBITS`.  Are you cool with leaving it as is?

**Comment by papertigers** [comment]
Note this is on hold until I can draft a slightly different solution:  lx should probably map the effective set to the limited set.

**Comment by citrus-it** [comment]
> Note this is on hold until I can draft a slightly different solution: lx should probably map the effective set to the limited set.

The problem is, AIUI, the Linux implementation is all at the thread level, whereas illumos privileges apply to the entire process.
I looked briefly into mapping this to native privileges but in the end this has been merged to OmniOS as-is.

File: usr/src/uts/common/brand/lx/syscall/lx_prctl.c
Comment (danmcd) [inline usr/src/uts/common/brand/lx/syscall/lx_prctl.c:287]: We should perhaps add @citrus-it 's comment in here as well:

"The problem is, AIUI, the Linux implementation is all at the thread level, whereas illumos privileges apply to the entire process."

Otherwise, @citrus-it is yanking it in to OmniOS, so perhaps we should as well?

**Comment by danmcd** [comment]
@citrus-it --> Do you have testing results to mention here?

**Review by danmcd — APPROVED** [review]
Adding an approval on behalf of @citrus-it (since he took it for OmniOS).

**Comment by danmcd** [comment]
Will add `integration-approval` once I see test results.

**Comment by papertigers** [comment]
Trying to page this back into memory.  https://www.kernel.org/doc/html/latest/userspace-api/no_new_privs.html says:
 `Once the bit is set, it is inherited across fork, clone, and execve and cannot be unset.`
And from a quick scan of the linux prctl man page it looks like `PR_CAPBSET_DROP` is what is thread aware?  Is that right?

I am fine with just merging this as is because it will unblock a lot of systemd services that now include NoNewPrivs.

As for testing, I don't have anything at hand anymore but when I originally posted the PR I verified the systemd service came up okay.

**Pull request merged**
